### PR TITLE
[CARBONDATA-3558] Clean up codes for property "autoRefreshDataMap"

### PR DESCRIPTION
**Pull request by kevinjmh**
Since PR #2255 support DEFERRED REBUILD when creating DataMap, autoRefreshDataMap property is not used any more.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/706/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/712/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/713/

File: core/src/main/java/org/apache/carbondata/core/datamap/DataMapProvider.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public void initData() { }
   /**
    * Rebuild the datamap by loading all existing data from mainTable
    * This is called when refreshing the datamap when
-   * 1. after datamap creation and if `autoRefreshDataMap` is set to true
+   * 1. after datamap creation and no "WITH DEFERRED REBUILD" defined
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datamap/DataMapProvider.java]: Please modify the MD document also
https://github.com/apache/carbondata/blob/master/docs/datamap/datamap-management.md#automatic-refresh

Comment (kevinjmh) [reply core/src/main/java/org/apache/carbondata/core/datamap/DataMapProvider.java]: The document does not mention `autoRefreshDataMap` now. As for the section title `Automatic Refresh`, it is explained  with `WITH DEFERRED REBUILD` in its parent section `DataMap Management`.


https://github.com/apache/carbondata/blob/2db8bf234b4234a9cbf3245ac39ed7391a84f77e/docs/datamap/datamap-management.md#L56-L64

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**